frontend/packages/kubevirt-plugin/src/models.ts

```diff
@@ -0,0 +1,162 @@
+import { K8sKind } from '@console/internal/module/k8s';
+import { TEMPLATE_TYPE_LABEL } from 'kubevirt-web-ui-components';
+
+export const VirtualMachineModel: K8sKind = {
+  label: 'Virtual Machine',
+  labelPlural: 'Virtual Machines',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachines',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachines',
+  abbr: 'VM',
+  namespaced: true,
+  kind: 'VirtualMachine',
+  id: 'virtualmachine',
+};
+
+export const VirtualMachineInstanceModel: K8sKind = {
+  label: 'Virtual Machine Instance',
+  labelPlural: 'Virtual Machine Instances',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachineinstances',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachineinstances',
+  abbr: 'VMI',
+  namespaced: true,
+  kind: 'VirtualMachineInstance',
+  id: 'virtualmachineinstance',
+};
+
+export const VirtualMachineInstancePresetModel: K8sKind = {
+  label: 'Virtual Machine Instance Preset',
+  labelPlural: 'Virtual Machine Instance Presets',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachineinstancepresets',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachineinstancepresets',
+  abbr: 'VMIP',
+  namespaced: true,
+  kind: 'VirtualMachineInstancePreset',
+  id: 'virtualmachineinstancepreset',
+};
+
+export const VirtualMachineInstanceReplicaSetModel: K8sKind = {
+  label: 'Virtual Machine Instance Replica Set',
+  labelPlural: 'Virtual Machine Instance Replica Sets',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachineinstancereplicaset',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachineinstancereplicasets',
+  abbr: 'VMIRS',
+  namespaced: true,
+  kind: 'VirtualMachineInstanceReplicaSet',
+  id: 'virtualmachineinstancereplicaset',
+};
+
+export const TemplateModel: K8sKind = {
+  label: 'Template',
+  labelPlural: 'Templates',
+  apiVersion: 'v1',
+  path: 'templates',
+  apiGroup: 'template.openshift.io',
+  plural: 'templates',
+  namespaced: true,
+  abbr: 'Template',
+  kind: 'Template',
+  id: 'template',
+};
+
+/* TODO(mlibra): migrate templates */
+export const VmTemplateModel: K8sKind = {
+  label: 'Virtual Machine Template',
+  labelPlural: 'Virtual Machine Templates',
+  apiVersion: 'v1',
+  path: 'templates',
+  apiGroup: 'template.openshift.io',
+  plural: 'templates',
```

Comment:
This should be:
`plural: 'vmtemplates'` then the menu item link is `/vmtemplates` but the page it'self stay `/tempates` and we get a `404` page :-( 

@mareklibra what am I missing ?

Reply:
Regarding the `404`, the `VmTemplateModel` is filtered out in `k8s-models.ts` on an attempt to redefine `Template` model.

@vojtechszocs , do you have any idea how to ease this check to enable specialization of a model?

If we stay strict on disabling redefinition, we might introduce explicit extension for `Router` so we will not land on the `VmTemplatesList` page via generic `/k8s/ns/:ns/:plural` but a specific one will be provided.